### ISSUE TITLE
Staging lines doesn’t stage

### ISSUE THREAD

**Issue by phaazon** [extrawurst/gitui#1580]
**Describe the bug**
Pressing `s` or any mapped key on `diff_stage_lines` doesn’t stage anything. However, it does unstage.

**To Reproduce**
1. Just try to stage some lines.

**Expected behavior**
It should stage.

**Screenshots**
N/A

**Context (please complete the following information):**
 - OS/Distro + Version: Linux awake 6.1.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 14 Feb 2023 22:08:08 +0000 x86_64 GNU/Linux
 - GitUI Version: `gitui 0.22.1`
 - Rust version: `rustc 1.66.1 (90743e729 2023-01-10)`

**Additional context**
N/A

**Comment by extrawurst** [comment]
can't reproduce:

![Screenflick Movie 8](https://user-images.githubusercontent.com/776816/222916985-021db20e-ee06-4cc6-a40d-f2e740378e16.gif)

**Comment by phaazon** [comment]
Okay, I’ll try again and will provide more information.

**Comment by phaazon** [comment]
I tried with a fresh build with `master`, and I still reproduce, whether I’m overriding the key or not.

**Comment by lightsnowball** [comment]
@phaazon is this maybe the case for new files in your repo? I reproduced such behavior with a new file (more precisely, a file that is untracked). 

@extrawurst is this expected for untracked files or should it be possible to stage such lines? I can make PR if needed.

**Comment by willparsons** [comment]
Just tried gitui for the first time and this threw me off at first.

It does seem like this only happens if a file is untracked, so you would have to stage the entire file first then un-stage the lines you don't want. This only happens the first time but is definitely annoying.

@extrawurst Does this sound like a bug or is it intentional?

**Comment by hadronized** [comment]
Yeah sorry I don’t really use `gitui` anymore (or actually any TUI anymore; most of them are too slow on super big monorepositories). However, I tried again and it seemed to work, so I guess something got fixed (I’m on a nightly).

And yeah, it doesn’t work on untracked files.
